api/python/quilt3/util.py

```diff
@@ -79,6 +79,116 @@ def __init__(self, message, **kwargs):
             setattr(self, k, v)
 
 
+class PhysicalKey(object):
+    __slots__ = ['bucket', 'path', 'version_id']
+
+    def __init__(self, bucket, path, version_id):
+        """
+        """
+        assert bucket is None or isinstance(bucket, str)
+        assert isinstance(path, str)
```

Comment:
We don't accept Path objects?

Reply:
The constructor is meant to be private, and other classes should only call `from_url` or `from_path`. But I don't know of a good way to enforce this.

Reply:
Ah ok. Maybe for now just add a docstring comment indicating that the constructor is supposed to be private and we can try to enforce it via code reviews.